### PR TITLE
Update actions

### PR DESCRIPTION
**Pull request by nehill197**
Added pkgdown action, updated code coverage action from private to public, and the R-CMD-check action.
Also updated the code coverage badge in README.

### CODE REVIEW

File: .github/workflows/test-coverage.yaml [hunk 1]
```diff
@@ -6,16 +6,16 @@ on:
...
+
+permissions: read-all
 
 jobs:
   test-coverage:
     runs-on: ubuntu-latest
     env:
-      GITHUB_PAT: ${{ secrets.PRIVATE_ACTIONS_PAT }}
-      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (aylapear) [inline .github/workflows/test-coverage.yaml:17]: This needs to remain.

File: _pkgdown.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+url: https://poissonconsulting.github.io/bisonpictools/
```
Comment (aylapear) [inline _pkgdown.yml:1]:
```suggestion
url: https://poissonconsulting.github.io/bisonpictools/
template:
  bootstrap: 5
```

**Review by aylapear — CHANGES_REQUESTED** [review]

File: .github/workflows/test-coverage.yaml (hunk 1 above)
Comment (nehill197) [reply .github/workflows/test-coverage.yaml:17]: Decided to try without these lines as that is what the template is currently giving us._

**Review by aylapear — APPROVED** [review]
Looks good

**Review by aylapear — APPROVED** [review]
Looks good

**Pull request merged**